### ISSUE TITLE
Use query cache when a query shares most of the query pipeline.

### ISSUE THREAD

**Issue by alexey-milovidov** [ClickHouse/ClickHouse#57490]
There are two ways to approach this task: easy and hard.

1. Easy.

When searching inside the query cache, also try to find the query after removal of LIMIT; LIMIT BY; ORDER BY; GROUP BY; WHERE. If found, construct a query pipeline that will use the data from the cache but apply the missing transformations.

2. Hard.

Serialize the query plan and put it into the cache instead of the AST hash. Search for incomplete query plans that could be completed.

Obviously, we should choose the easy way.

**Comment by pashandor789** [comment]
should I create `Planner` in executeQuery.cpp and use it? Or should I add these transformations outside of the `Planner`?

**Comment by cleaton** [comment]
Would it make sense to do something similar for query condition cache? Use heuristics to find common subset of filters to cache in LRU/LFU. (Ex hash per filter or filter group rather than all filters together as one hash)

**Comment by alexey-milovidov** [comment]
@cleaton Yes, but I expect the benefits will be limited w.r.t. the implementation complexity.

**Comment by rschu1ze** [comment]
Related: https://github.com/ClickHouse/ClickHouse/issues/93756

**Comment by xuir268** [comment]
@rschu1ze , can I be assign for this issue

**Comment by rschu1ze** [comment]
We plan to look into this issue soon by ourselves.

**Comment by xuir268** [comment]
okay , meanwhile should i raise any implementation which may help or something
